src/app/3d/qgsmaptoolclippingplanes.h

```diff
@@ -0,0 +1,64 @@
+/***************************************************************************
+    qgsmaptoolclippingplanes.h
+    ---------------------
+    begin                : March 2025
+    copyright            : (C) 2025 by Matej Bagar
+    email                : matej dot bagar at lutraconsulting dot co dot uk
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSMAPTOOLCLIPPINGPLANES_H
+#define QGSMAPTOOLCLIPPINGPLANES_H
+
+#include "qgsmaptool.h"
+#include "qgspointxy.h"
+#include "qgsrubberband.h"
+#include "qobjectuniqueptr.h"
+
+
+class Qgs3DMapCanvasWidget;
+class QgsMapCanvas;
+
+
+/**
+ * \ingroup app
+ * \brief A map tool that stores clipping planes from lines drawn onto map canvas.
```

Comment:
nitpick: this tool does not "store" the clipping planes, it allows to "define" them.